### DISCUSSION TITLE
Unable to Read in Data From IR Seeker

### DISCUSSION THREAD

**Discussion by ItzBerkPower** [pybricks/support#1104] [Q&A] [answered]
```python
from pybricks.iodevices import PUPDevice
from pybricks.parameters import Port

device = PUPDevice(Port.A)

while True:
    values = device.read(5)
    if len(values) > 0:
        print("Strength", values[0]) #return the signal strength
    if len(values) > 1:
        print("Direction", values[1]) #return the direction of the ball 0 - 12
```
This code is meant to be able to read in the IR Ball data from the IR Seeker, but it isn't printing anything to the terminal.

**Comment by laurensvalk** [comment]
We've had a few support tickets for this one (including a team that was in contact with BBS), but unfortunately we don't have this device ourselves.

It would be helpful if BBS got in touch with us directly at team@pybricks.com. If you try their customer service, please let them know. Thanks!

**Reply by ItzBerkPower** [reply]
Oh ok, thank you!

Though on further inspection, I tried running;

```python
print("hi")
```

and nothing was printed to the Terminal (The code was run on its own).

**Reply by laurensvalk** [reply, marked as answer]
It sounds like you might have installed the firmware via https://beta.pybricks.com/ but are using https://code.pybricks.com/ to write code.

If you install the beta firmware, be sure to use https://beta.pybricks.com/ for coding too.
